03_module/hello.c

```diff
@@ -0,0 +1,27 @@
+#include <linux/module.h>
+#include <linux/moduleparam.h>
+#include <linux/init.h>
+#include <linux/kernel.h>
+
+int a = 0; 
+int b = 0;
+
+module_param(a, int, 0660);
+module_param(b, int, 0660);
+
+int hello_init(void)
+{
+	printk(KERN_WARNING "\na + b = %i\n", a + b);
+	return 0;
+}
+
+void hello_exit(void)
+{
+	printk(KERN_WARNING "\na - b = %i\n", (a < b)?(b - a):(a - b));
+
+}
+
+MODULE_LICENSE("GPL");
+
+module_init(hello_init);
+module_exit(hello_exit);
```

Comment:
WARNING: Missing or malformed SPDX-License-Identifier tag in line 1
#1: FILE: /air/Documents/trainings/linux/kernel/pro-camp/2021/03_module/hello.c:1:
+#include <linux/module.h>

ERROR: trailing whitespace
#6: FILE: /air/Documents/trainings/linux/kernel/pro-camp/2021/03_module/hello.c:6:
+int a = 0; $

ERROR: do not initialise globals to 0
#6: FILE: /air/Documents/trainings/linux/kernel/pro-camp/2021/03_module/hello.c:6:
+int a = 0; 

ERROR: do not initialise globals to 0
#7: FILE: /air/Documents/trainings/linux/kernel/pro-camp/2021/03_module/hello.c:7:
+int b = 0;

WARNING: Prefer [subsystem eg: netdev]_warn([subsystem]dev, ... then dev_warn(dev, ... then pr_warn(...  to printk(KERN_WARNING ...
#14: FILE: /air/Documents/trainings/linux/kernel/pro-camp/2021/03_module/hello.c:14:
+	printk(KERN_WARNING "\na + b = %i\n", a + b);

WARNING: Prefer [subsystem eg: netdev]_warn([subsystem]dev, ... then dev_warn(dev, ... then pr_warn(...  to printk(KERN_WARNING ...
#20: FILE: /air/Documents/trainings/linux/kernel/pro-camp/2021/03_module/hello.c:20:
+	printk(KERN_WARNING "\na - b = %i\n", (a < b)?(b - a):(a - b));

WARNING: adding a line without newline at end of file
#27: FILE: /air/Documents/trainings/linux/kernel/pro-camp/2021/03_module/hello.c:27:
+module_exit(hello_exit);